lib/swoosh/adapters/mailgun.ex

```diff
@@ -21,17 +21,47 @@ defmodule Swoosh.Adapters.Mailgun do
         use Swoosh.Mailer, otp_app: :sample
       end
 
+  ## Using with provider options
+
+      import Swoosh.Email
+
+      new()
+      |> from({"T Stark", "tony.stark@example.com"})
+      |> to({"Steve Rogers", "steve.rogers@example.com"})
+      |> to("wasp.avengers@example.com")
+      |> reply_to("office.avengers@example.com")
+      |> cc({"Bruce Banner", "hulk.smash@example.com"})
+      |> cc("thor.odinson@example.com")
+      |> bcc({"Clinton Francis Barton", "hawk.eye@example.com"})
+      |> bcc("beast.avengers@example.com")
+      |> subject("Hello, Avengers!")
+      |> html_body("<h1>Hello</h1>")
+      |> text_body("Hello")
+      |> put_provider_option(:custom_vars, %{"key" => "value"})
+      |> put_provider_option(:recipient_vars, %{"steve.rogers@example.com": %{var1: 123}, "juan.diaz@example.com": %{var1: 456}})
+      |> put_provider_option(:sending_options, %{dkim: "yes", tracking: "no"})
+      |> put_provider_option(:tags, ["worldwide-peace", "unity"])
+      |> put_provider_option(:template_name, "avengers-templates")
+
   ## Provider options
 
-  - `:custom_vars` (used to translate to `v:`, now `h:X-Mailgun-Variables`)
-  - `:sending_options` (`o:`)
-  - `:template_name` (template)
-  - `:recipient_vars` (recipient-variables)
-  - `:tags` (`o:tag`, was added in before `:sending_options`)
+    * `:custom_vars` (map) - used to translate to `v:my-var`, now
+      `h:X-Mailgun-Variables`, add custome data to email
+
+    * `:recipient_vars` (map | list[map]) - `recipient-variables`, assign
```

Comment:
silly question (I haven't used mailgun in a while), what happens when it's a list? how does it map recipients to variables?

Reply:
I was having the same thoughts as well, assumption of the type for `recipient_vars` made based on these codes:
```
191   defp encode_variable(var) when is_map(var) or is_list(var),
192     do: Swoosh.json_library().encode!(var)
193 
194   defp encode_variable(var), do: var
```

See https://github.com/swoosh/swoosh/blob/main/lib/swoosh/adapters/mailgun.ex#L191

Reply:
https://github.com/swoosh/swoosh/pull/418 I guess list was there to be flexible, but recipient vars should probably be a map.

Reply:
```suggestion
    * `:recipient_vars` (map) - `recipient-variables`, assign
```